src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,235 @@
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todos: Todo[];
+  filter: string;
+  changeFilter: (string: string) => void;
+  tempTodo: Todo | null;
+  deleteTodo: (number: number) => void;
+  deleteCompleted: () => void;
+}
+
+export const TodoList: React.FC<Props> = ({
+  todos, filter, changeFilter, tempTodo, deleteTodo, deleteCompleted,
+}) => {
+  const todosRemaining = todos.reduce((acc, val) => {
+    if (!val.completed) {
+      return acc + 1;
+    }
+
+    return acc;
+  }, 0);
+
+  const [deletionOverlay, setDeletionOverlay] = useState(0);
+
+  return (
+    <>
+      <section className="todoapp__main" data-cy="TodoList">
+        {todos.map(todo => (
+          <div
+            data-cy="Todo"
+            className={cn('todo', { completed: todo.completed })}
+            key={todo.id}
+          >
+            <label className="todo__status-label">
+              <input
+                data-cy="TodoStatus"
+                type="checkbox"
+                className="todo__status"
+                checked={todo.completed}
+              />
+            </label>
+
+            <span data-cy="TodoTitle" className="todo__title">
+              {todo.title}
+            </span>
+
+            {/* Remove button appears only on hover */}
+            <button
+              type="button"
+              className="todo__remove"
+              data-cy="TodoDelete"
+              onClick={() => {
+                setDeletionOverlay(todo.id);
+                deleteTodo(todo.id);
+              }}
+            >
+              ×
+            </button>
+
+            {/* overlay will cover the todo while it is being updated */}
+            <div
+              data-cy="TodoLoader"
+              className={cn('modal overlay',
+                { 'is-active': deletionOverlay === todo.id })}
+            >
+              <div className="modal-background has-background-white-ter" />
+              <div className="loader" />
+            </div>
+          </div>
+        ))}
+
+        {tempTodo
+        && (
+          <div data-cy="Todo" className="todo">
+            <label className="todo__status-label">
+              <input
+                data-cy="TodoStatus"
+                type="checkbox"
+                className="todo__status"
+              />
+            </label>
+
+            <span data-cy="TodoTitle" className="todo__title">
+              {tempTodo?.title}
+            </span>
+
+            <button type="button" className="todo__remove" data-cy="TodoDelete">
+              ×
+            </button>
+
+            {/* 'is-active' class puts this modal on top of the todo */}
+            <div data-cy="TodoLoader" className="modal overlay is-active">
+              <div className="modal-background has-background-white-ter" />
+              <div className="loader" />
+            </div>
+          </div>
+        )}
+
+        {/* This is a completed todo */}
+
+        {/* This todo is not completed */}
```

Comment:
Don't forget to remove commented out lines

Reply:
I will do so when completing third task to not delete crucial information